File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/SalaryGradeRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface SalaryGradeRow {
    GradeId?: number;
    GradeName?: string;
    GradeCode?: string;
    MinSalary?: number;
    MaxSalary?: number;
    Description?: string;
    IsActive?: boolean;
}

export abstract class SalaryGradeRow {
    static readonly idProperty = 'GradeId';
    static readonly nameProperty = 'GradeName';
    static readonly localTextPrefix = 'Operations.SalaryGrade';
    static readonly lookupKey = 'Operations.SalaryGrade';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<SalaryGradeRow>('Operations.SalaryGrade') }
    static async getLookupAsync() { return getLookupAsync<SalaryGradeRow>('Operations.SalaryGrade') }

    static readonly deletePermission = 'Operations:Payroll';
    static readonly insertPermission = 'Operations:Payroll';
    static readonly readPermission = 'Operations:Payroll';
    static readonly updatePermission = 'Operations:Payroll';

    static readonly Fields = fieldsProxy<SalaryGradeRow>();
}

File: HRMS/HRMS.Web/Modules/Operations/SalaryGrade/SalaryGradeRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace HRMS.Operations;

[ConnectionKey("Default"), Module("Operations"), TableName("SalaryGrades")]
[DisplayName("Salary Grades"), InstanceName("Salary Grade")]
[ReadPermission("Operations:Payroll")]
[ModifyPermission("Operations:Payroll")]
[LookupScript]
public sealed class SalaryGradeRow : Row<SalaryGradeRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Grade Id"), Identity, IdProperty]
    public int? GradeId { get => fields.GradeId[this]; set => fields.GradeId[this] = value; }

    [DisplayName("Grade Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string GradeName { get => fields.GradeName[this]; set => fields.GradeName[this] = value; }

    [DisplayName("Grade Code"), Size(50)]
    public string GradeCode { get => fields.GradeCode[this]; set => fields.GradeCode[this] = value; }

    [DisplayName("Min Salary"), Size(19), Scale(2)]
    public decimal? MinSalary { get => fields.MinSalary[this]; set => fields.MinSalary[this] = value; }

    [DisplayName("Max Salary"), Size(19), Scale(2)]
    public decimal? MaxSalary { get => fields.MaxSalary[this]; set => fields.MaxSalary[this] = value; }

    [DisplayName("Description"), Size(500)]
    public string Description { get => fields.Description[this]; set => fields.Description[this] = value; }

    [DisplayName("Is Active"), NotNull, DefaultValue(true)]
    public bool? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field GradeId;
        public StringField GradeName;
        public StringField GradeCode;
        public DecimalField MinSalary;
        public DecimalField MaxSalary;
        public StringField Description;
        public BooleanField IsActive;
    }
}
